.github/workflows/continous-delivery.yml

```diff
@@ -1,5 +1,12 @@
 name: CD
 
+# Ensures that only one deploy task will run at a time. Previous deployments, if
+# already started, won't get cancelled. Following workflow runs will get queued
+# until the previous deployment(s) finishes
```

Comment:
This config has acceptable behaviour, but these comments don't quite explain how it works. Interim pending runs get still get cancelled, only the first and latest run will complete:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency